js/src/data/utils.test.js

```diff
@@ -65,4 +73,180 @@ describe( 'getReportQuery', () => {
 			).toHaveProperty( 'orderby', freeFields[ 0 ] );
 		} );
 	} );
+
+	it( 'When the query does not contain `programs` or `products`, should have no `ids` field', () => {
+		forAnyCategoryAndDateReference( ( category, dateReference ) => {
+			const query = {};
+
+			expect(
+				getReportQuery( category, 'free', query, dateReference ).ids
+			).toBeUndefined();
+		} );
+	} );
+
+	it( 'When the programs report query contains `programs`, should have `ids` field', () => {
+		const query = {
+			programs: '123,456',
+		};
+
+		expect(
+			getReportQuery( 'programs', 'free', query, 'primary' ).ids
+		).toBe( query.programs );
+	} );
+
+	it( 'When the products report query contains `products`, should have transformed `ids` field', () => {
```

Comment:
I'd explain what "transformation" is expected.
```suggestion
	it( 'When the products report query contains `products`, should have `ids` field, with values prepended with `gla_`', () => {
```